uiowa/uiowa
# Upgrade to Drupal 10.0

Resolves #6959 
Resolves #6537 
Resolves #5738
Resolves #5454
Resolves #6566

# ToDo

- [x] Update tippie split once https://github.com/uiowa/uiowa/pull/7033 merges with the robots.txt changes
- [x] Fix layout builder settings tray regression
- [x] Resolve CCOM split issue - _https://github.com/uiowa/uiowa/pull/6994#issuecomment-1777826089_

# Explanation of some of the configuration differences

- Some sites had rabbit hole before default had it, so this PR cleans those instances up.
- Admissions, the events categories taxonomy was moved to the feature so the site split no longer needs those permissions.
- Robots.txt was updated based on a core change to hide oembed routes.
- ighn.international permission clean up. v2 permissions related to blocks take precedence.
- Some modules updated and provide a new setting or replace an existing

# How to test
- Check to see if update to typhonius/acquia-php-sdk-v2 can close https://github.com/uiowa/uiowa/issues/5738
  - `./vendor/bin/blt uiowa:multisite:create foo.uiowa.edu --no-db --no-commit` - See no early warnings in the command. Ctrl+C to exit.
- Check to see if update to drupal/redirect solves https://github.com/uiowa/uiowa/issues/1072 (I know the linked upstream issue is still out there, but I can't reproduce the issue)
- A patch, "Support third party settings for components within a section," was updated related to https://www.drupal.org/project/drupal/issues/3015152
  - This was introduced to make our unique id field on blocks a reality and we have run with it since. To test, make sure you are still able to see/add/edit unique ids on our blocks. 
- A patch, "Changing an existing embedded media's alignment or alt data attributes does not get saved with CKEditor," was removed in favor of the now contrib CKEditor module's version https://www.drupal.org/project/drupal/issues/3102249 -> https://www.drupal.org/project/ckeditor/issues/3330723
  - To test, go to https://sandbox.prod.drupal.uiowa.edu/node/251/edit and make sure that when you change alignment for the media item, it retains your selection after save.
- A patch, "Hidden buttons in off-canvas dialog are not being hidden," was removed related to https://www.drupal.org/project/drupal/issues/3070375
  - A CSS rule was added to fix the button showing up.
  - To test, with the media library widget, there is a button that is supposed to be hidden labeled "Update widget," and it gosh darn better stay that way.
- A patch, "Expose Layout Builder data to REST and JSON:API" was updated related to https://www.drupal.org/project/drupal/issues/2942975
  - This was introduced to provide layout builder flavored default content. To test, one should run a site install (it should be successful) and try exporting the default content again, there should be no major changes besides revision numbers and such.
- A patch, "ImageItem::getUploadValidators() should be the source of truth for validating uploaded images," was updated related to https://www.drupal.org/project/drupal/issues/3008292
  - This was introduced to enforce max image sizes. Media was skipping over it. To test, upload an image beyond our max image size. You should get the message it was sized down, but now inspect the original that was uploaded it should have actually been sized down.
- A patch, "layout_builder__layout_section column hitting database limit," was updated related to https://www.drupal.org/project/drupal/issues/3030154
  - See if you can break it. Or not. Might not be worth testing beyond it applying successfully.
- A patch, "Logs flooded with warning messages Invalid display settings encountered," was removed related to https://www.drupal.org/project/entity_embed/issues/3077225
  - No testing needed as the module was removed.
